harmonica/filters/_filters.py

```diff
@@ -62,3 +64,599 @@ def derivative_upward_kernel(fft_grid, order=1):
+        *freq_northing*, *freq_easting*.
+        Use :func:`xrft.xrft.fft` and :func:`xrft.xrft.ifft` functions to
+        compute the Fourier Transform and its inverse, respectively.
+    inclination : float in degrees
+        The inclination of the inducing Geomagnetic field. Default is 90 degree
+         for RTP field.
+    declination : float in degrees
+        The declination of the inducing Geomagnetic field. Default is 0 degree
+         for RTP field.
+    magnetization_inclination : float in degrees or None
+        The inclination of the total magnetization of the anomaly source. If
+        None, the ``magnetization_inclination`` will be set equal to the
+        ``inclination``, neglecting remanent magnetization and self
+        demagnetization. Default None.
+    magnetization_declination : float in degrees
+        The declination of the total magnetization of the anomaly source. If
+        None, the ``magnetization_declination`` will be set equal to the
+        ``declination``, neglecting remanent magnetization and self
+        demagnetization. Default None.
+    ambient_field : float or 2d-array
+        Ambient field in the study area. It can use the mean ambinent field
+        value in the study area or the real ambient field value in all
+        locations. Default is 50,000 nT.
+
+    Returns
+    -------
+    da_filter : :class:`xarray.DataArray`
+        Array with the kernel for the pseudo gravity filter in frequency
+        domain.
+
+    References
+    ----------
+    [Salem2014]_
+
+    See also
+    --------
+    harmonica.pseudo_gravity
+    """
+    # Check if magnetization angles are valid
+    _check_magnetization_angles(magnetization_inclination, magnetization_declination)
+    # Define magnetization angles if they are None
+    if magnetization_declination is None and magnetization_inclination is None:
+        magnetization_inclination = inclination
+        magnetization_declination = declination
+    # Catch the dims of the Fourier transformed grid
+    dims = fft_grid.dims
+    # Grab the coordinates of the Fourier transformed grid
+    freq_easting = fft_grid.coords[dims[1]]
+    freq_northing = fft_grid.coords[dims[0]]
+    # Convert frequencies to wavenumbers
+    k_easting = 2 * np.pi * freq_easting
+    k_northing = 2 * np.pi * freq_northing
+    # Calculate vertical integral
+    da_filter = np.sqrt(k_easting**2 + k_northing**2) ** -1
+    # Check if input is RTP field
+    if inclination != 90:
+        # Add RTP kernel to the calculate vertical integral
+        da_filter *= _get_rtp_filter(
+            k_easting,
+            k_northing,
+            inclination,
+            declination,
+            magnetization_inclination,
+            magnetization_declination,
+        )
+    # Set 0 wavenumber to 0
+    da_filter.loc[dict(freq_northing=0, freq_easting=0)] = 0
+    return da_filter * GRAVITATIONAL_CONST * 1e8 / ambient_field
```

Comment:
I think this looks much better than before. Although I'm still unsure where the `1e8` is coming from. I suspect there's some portion of it that might be due to the fact that we need to convert the `ambient_field` from nT to T, but I'm still one order of magnitude short haha.

Reply:
Haha. It's a bit confusing... 
If we look at the formula 9:
![image](https://user-images.githubusercontent.com/54405391/209063116-faccf7ec-c5cb-4ef7-9ed9-20298b46e412.png)
Where VI in `nT*m,` g in `mGal,` p in `g*cm-3,` F in `nT,` and G in `m3*kg−1*s−2`
So convert from `mGal` to `m*s-2`, there is a `1e-5,` and p from `g*cm-3` to `kg*m3` with a `1e3`
So VI/g=`1e-5`/`1e3` * `(k*F)/(p*G)`=(`1e-8/G`)*`F*k/p`
Thus `pseudogravity= VI * G * 1e8 / F`

Reply:
Is that make sense to you? Haha

Reply:
Thanks for the clarification @LL-Geo. I think I understand where the `1e-8` comes from now. But I still have many doubts on the development of the equations in that paper in comparison with Blakely.

For example, the authors in equation 3 assumes that there's no remanent magnetization in the source (magnetization is parallel to the ambient field), but in our implementation of the filter we do allow passing different angles for the magnetization and for the geomagnetic field.

Also, they explicitly mention that the susceptibility is given in cgs units, so I suspect that there might be another unit conversion missing. Moreover, is there a reason why the filter don't ask values for the density and the susceptibility? Why assuming that the ratio between them is equal to 1?

I think the pseudogravity field requires some discussion, not only in its implementation but also on its design. So here's my idea: what if we remove it from this PR and we can include it in a separate one? This way we don't block all the other filters (that are pretty ready to be merged) just because we have some discussions on a single one of them.

Let me know what do you think.

Reply:
Haha. Both ways works!
The Blakely's implementation is confusing as well. If you have the book and look at the page 408.
![image](https://user-images.githubusercontent.com/54405391/209262743-ac0cb43c-175d-4cd9-a856-9a3439e8309b.png)
![image](https://user-images.githubusercontent.com/54405391/209263047-060d3955-7e28-4fe2-bbab-10c598efee69.png)

That implementation is based on 100 kg/m3 per A/m.
The idea of the new implementation is decoding magnetization into the ambient field and also magnetic susceptibility.
I know that is not ideal, because we have remanent and demagnetization. So I put the option to include different magnetization inclinations and declinations. That could refer to as apparent susceptibility. If we don't have remanent, everything is perfect (apparent susceptibility equal to the real susceptibility). But sometimes, we need to have some assumptions in the real world.

 As for the unit transformation. For cgs unit, μ=1, as for SI unit, we need to add a 4 pi in here. So the 1 susceptibility(cgs)= 4 Pi susceptibility(SI).

Reply:
Oh yes, those FORTRAN implementations are not the best, we aim to do something better than that 😎 .

> The idea of the new implementation is decoding magnetization into the ambient field and also magnetic susceptibility.
I know that is not ideal, because we have remanent and demagnetization. So I put the option to include different magnetization inclinations and declinations.

I agree with this! I don't like that a lot of geophysical applications assume that susceptibility is a good representative of magnetization of the sources on its own: remanence do exist in nature and it's more common that we usually assume. The point is that Salem's paper explicitly assumes no remanence in equation 3, where they consider that the magnetization vector is parallel to the ambient field:

$$ \mathbf{J} = k \mathbf{F} $$

or with a notation I'm more familiar with:

$$ \mathbf{M} = \chi \mathbf{F} $$

So their equations wouldn't be compatible with a pseudogravity that do takes remanence into account, am I right?

> As for the unit transformation. For cgs unit, μ=1, as for SI unit, we need to add a 4 pi in here. So the 1 susceptibility(cgs)= 4 Pi susceptibility(SI).

Nice!

Again, I think the pseudogravity filter deserves some more discussion and I wouldn't like to block this PR just because of it. So I retake my suggestion of removing it from this PR and open a new one just for it.

Reply:
Ahaha.. Let me explain in this way.
In the real world... We have remanent, and we can measure it and calculate the Koenigsberger Ratio (Q). So we have: 
M=xF + M(remanent) = xF + Q*xF = (1+Q)*x*F
I will call (1+Q)*x as the apparent susceptibility.
So. If no remanent Q is 0, the apparent susceptibility is the real susceptibility. If Q is not 0, what should we do?
The answer is petrophysics!

If we don't know petrophysics, we don't know magnetization (although we have some numerical way for approximation), and Koenigsberger Ratio. I guess that's out of the scope of this filter. Because this filter is just RTP plus vertical integral. And we try to link some physics in here.

I will add some lines in the doc to show the unit of this filter (g/cm-3 per apparent susceptibility). And if they have petrophysics (with Q), they can calculate the real susceptibility on their own!

Let me know how do you think (if you still think we need more discussion, I'll remove pg and open a new PR)! And thanks so much for this @santisoler 👍 😄

Reply:
Thanks for explanation @LL-Geo!

I see your point. Although I have to say that the remanence that can be considered when Q is non-zero is remanent magnetization that is parallel to the geomagnetic field. In the general case this doesn't happen: rocks can present magnetization on other directions than $\mathbf{F}$, so we cannot express the magnetization vector $\mathbf{M}$ as a scalar times $\mathbf{F}$.

I would love to continue this discussion on the pseudogravity filter. But at the same time I would like to release the next version of Harmonica this month and I'm thrilled to include the FFT in it.
So I would take your proposal of stripping the PG filter from this PR and open a new one for it.

Reply:
Okay dokie! Let's move forward @santisoler !

I have removed the PG filter. It should be ready to go 😄

Once this one is merged, I'll open a new PR for PG to continue our discussion!

Reply:
is thefe one?